### ISSUE TITLE
Installation Fails with Ruby 2.5.1

### ISSUE THREAD

**Issue by bowmanat** [patrickdavey/vimwiki_markdown#14]
Even when using rvm, I was unable to install the gem due to an issue installing rugged.  It seems that this is an upstream issue -- I found another project that has the same problem (https://github.com/samaaron/sonic-pi/issues/1908).

To get around this, I used rvm to install Ruby 2.3.3, and everything seems to work.  For anyone like me who is unfamiliar with Ruby, I used

```
rvm install 2.3.3
rvm ruby-2.3.3 do gem install vimwiki_markdown
```

You probably also need to make 2.3.3 your default.  I just removed all other versions of Ruby instead.

**Comment by maikeldotuk** [comment]
Any change for this to work on the latest Ruby or is this repo unmaintained?

**Comment by patrickdavey** [comment]
Weird, I must have missed this issue! (my daughter was born on the 18th Sept which is probably why)

I'll try to make some time in the next few weeks to look into this. If you have the time @maikeldotuk / @bowmanat to try fix it yourself, PRs are welcome :)

**Comment by patrickdavey** [comment]
I just used `rvm` with `ruby-2.5.1` and built the gem fine locally, and installed it and ran it against a vimwiki install (this is with the current version of `vimwiki_markdown`)

So, I'm not sure what the issue is that you're having. Can you please provide steps to reproduce whatever error you're getting.  Unfortunately all I can say at the moment is "it works fine for me under ruby-2.5.1" ... which may not be much use, but, it's still true ;)

**Comment by maikeldotuk** [comment]
After javascript, rust and Python I don't have time for another new language On 7 Mar 2019 20:02, Patrick Davey <notifications@github.com> wrote:Weird, I must have missed this issue! (my daughter was born on the 18th Sept which is probably why)
I'll try to make some time in the next few weeks to look into this. If you have the time @maikeldotuk / @bowmanat to try fix it yourself, PRs are welcome :)

**Comment by patrickdavey** [comment]
Fair enough.  Well, you have a workaround (use ruby-2.3.3 as suggested in https://github.com/patrickdavey/vimwiki_markdown/issues/14#issue-364605002). 

If you want me to look at 2.5.1 issue, then please provide an error report or something to go on, as I am having zero issues installing and using the gem under 2.5.1.

**Comment by maikeldotuk** [comment]
To be honest after I managed to install it following the guidelines of the user in the issue, it didn't work at all. It was issue after issue after issue after issue so I ended up writting my own code based on the code of another user. 

https://gist.github.com/maikeldotuk/54a91c21ed9623705fdce7bab2989742

On Thursday, March 07, 2019 23:09 GMT, Patrick Davey <notifications@github.com> wrote:
 
I just used rvm with ruby-2.5.1 and built the gem fine locally, and installed it and ran it against a vimwiki install (this is with the current version of vimwiki_markdown)
So, I'm not sure what the issue is that you're having. Can you please provide steps to reproduce whatever error you're getting. Unfortunately all I can say at the moment is "it works fine for me under ruby-2.5.1" ... which may not be much use, but, it's still true ;)

**Comment by patrickdavey** [comment]
Nice gist :) Pandoc certainly is a neat way to approach this, thanks for sharing!

Sorry you had so many issues, without being able to reproduce the problem you're having it's obviously hard to fix.

Still, great you've got a solution working :)

If anyone else comes across this thread, if you could produce an error trace that'd be most helpful.

**Comment by bowmanat** [comment]
> Weird, I must have missed this issue! (my daughter was born on the 18th Sept which is probably why)

Congrats!

> If anyone else comes across this thread, if you could produce an error trace that'd be most helpful.

Entire error message:
[wiki-install-error.txt](https://github.com/patrickdavey/vimwiki_markdown/files/2947113/wiki-install-error.txt)

gem_make.out:
[gem_make.txt](https://github.com/patrickdavey/vimwiki_markdown/files/2947131/gem_make.txt)

gcc version is 8.2.1 20181127

The error message in the logs is the same error as the one linked to in my first post.
[error link](https://github.com/samaaron/sonic-pi/issues/1908)

Possibly related link:
https://bugs.ruby-lang.org/issues/14425
Though I think the link is saying that the problem was fixed...

Anyway, looks to be more an issue with rugged, so not sure how much you can do unless you can remove the dependency.  But I'm not super familiar with Ruby's ecosystem.

**Comment by patrickdavey** [comment]
Thanks very much for the congrats ;) and the bug report.

I reworked the gem today and its now using (I think) the latest versions of the various Github/Markup gems etc. There's a serious lack of documentation around it!.

Anyway, it mostly seems to work, unfortunately it will quite possibly break peoples links (which was actually broken when we introduced the tag functionality back in https://github.com/patrickdavey/vimwiki_markdown/commit/0644ac831466158006b9c91a1b9b5e1c8ce3dc0a . Still, alls fair if I haven't released version 1.0 right ;)

The new version will remove the dependency on `rugged`, and, I hadn't even realised there was an underlying dependency on python from pygments, that should also be gone now.

Anyway, just a bit more testing in the next day or two and then I'll release the new version. I'll comment here when I've done that.

Thanks again :)
